test/library/waits.go

```diff
@@ -138,3 +139,32 @@ func WaitForHTTPStatus(t *testing.T, waitDuration time.Duration, client *http.Cl
 		return false, nil
 	})
 }
+
+func WaitForNewKASRollout(t *testing.T, ctx context.Context, kasClient operatorv1.KubeAPIServerInterface, origRevision int32) error {
+	err := wait.PollUntilContextTimeout(ctx, 10*time.Second, 30*time.Minute, true, func(ctx context.Context) (bool, error) {
+		kas, err := kasClient.Get(ctx, "cluster", metav1.GetOptions{})
```

Comment:
I think it's a better practice to pass the context into `WaitForNewKASRollout`, so that we can control its execution with the same context; i.e. if that context gets cancelled (or its parent), then the wait would also be cancelled. I am not sure why `TODO()` is used in the other tests instead of an actual test context.